### PR TITLE
Avoid duplicate bundle IDs

### PR DESCRIPTION
**Pull request by pvollan**
#### d19015d9f1ea1815dd30e272c1367d2c78f497cd
<pre>
Avoid duplicate bundle IDs
<a href="https://bugs.webkit.org/show_bug.cgi?id=267393">https://bugs.webkit.org/show_bug.cgi?id=267393</a>
<a href="https://rdar.apple.com/120823686">rdar://120823686</a>

Reviewed by NOBODY (OOPS!).

Avoid duplicate bundle IDs when building extensions. Duplicate bundle IDs can cause
issues with launching WebKit processes. This is fixed by changing the bundle IDs of
the XPC services, and reverting some changes that prevented the XPC services from
being built.

* Source/WebKit/Configurations/BaseXPCService.xcconfig:
* Source/WebKit/Configurations/GPUService.xcconfig:
* Source/WebKit/Configurations/NetworkService.xcconfig:
* Source/WebKit/Configurations/WebContentService.xcconfig:
</pre>
https://github.com/WebKit/WebKit/commit/d19015d9f1ea1815dd30e272c1367d2c78f497cd

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/33555 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/12328 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/35480 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/36173 "Built successfully") | [❌ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/30448 "Failed to checkout and rebase branch from PR 22643") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/14686 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/9485 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/29561 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/34030 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/10389 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/29919 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/9073 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/9171 "Passed tests") | | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/37500 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/30447 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/30255 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/35303 "Passed tests") | 
| | [❌ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/9306 "Failed to checkout and rebase branch from PR 22643") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/7254 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/33182 "Passed tests") | 
| | [❌ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/11071 "Failed to checkout and rebase branch from PR 22643") | | | 
| | [❌ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/9890 "Failed to checkout and rebase branch from PR 22643") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/10066 "Built successfully") | | |

### CODE REVIEW

File: Source/WebKit/Configurations/WebContentService.xcconfig [hunk 1]
```diff
@@ -30,6 +30,12 @@ PRODUCT_NAME_Development = com.apple.WebKit.WebContent.Development;
 PRODUCT_NAME_CaptivePortal = com.apple.WebKit.WebContent.CaptivePortal;
 
 PRODUCT_BUNDLE_IDENTIFIER = $(PRODUCT_NAME);
+PRODUCT_BUNDLE_IDENTIFIER[sdk=iphone*] = $(PRODUCT_NAME)Service;
+PRODUCT_BUNDLE_IDENTIFIER[sdk=iphone*17.0*] = $(PRODUCT_NAME);
+PRODUCT_BUNDLE_IDENTIFIER[sdk=iphone*17.1*] = $(PRODUCT_NAME);
+PRODUCT_BUNDLE_IDENTIFIER[sdk=iphone*17.2*] = $(PRODUCT_NAME);
+PRODUCT_BUNDLE_IDENTIFIER[sdk=iphone*17.3*] = $(PRODUCT_NAME);
```
Comment (brentfulgham) [inline Source/WebKit/Configurations/WebContentService.xcconfig:37]: Could these just be `sdk=iphone*17*]` instead?

**Review by brentfulgham — APPROVED** [review]
Looks reasonable! r=me

File: Source/WebKit/Configurations/WebContentService.xcconfig (hunk 1 above)
Comment (emw-apple) [reply Source/WebKit/Configurations/WebContentService.xcconfig:37]: These are intentionally _not_ matching the OS versions where we're using extensions.

**Review by emw-apple — CHANGES_REQUESTED** [review]
I've mentioned to Per Arne already in DMs, but I think this is the wrong approach, because of interactions with internal workflows. r-, here's my reasoning:

The problem being solved is that, during the XPC to EK migration, we need to simultaneously (1) install ExtensionKit bundles with the traditional WebKit process bundle IDs and (2) ensure that XPC services using those same bundle IDs are _not_ installed, to avoid ID collisions on the system.

The approach I approved and we landed for (2) was to skip installing XPC services when building for OS versions that have the EK migration. That satisfies the build, but it's not enough for _installing_ WebKit roots onto existing OS builds, with the XPC services (and their colliding bundle IDs) already present.

This PR does a switcheroo where we continue to install the old, unneeded XPC services, but we append "Service" to their bundle IDs to satisfy (2). I think this is the wrong approach because:
- The OS has caches that map bundle ID to binary path, so installing a root with these code changes is not enough to guarantee that extensions will be used, users have to clear their cache as well.
- It drags out the XPC to EK migration and will make it take multiple OS builds to complete. We'd need to land this, get it onto every internal OS build and device we care about, and only _then_ can we stop installing XPC services.
- Once the migration is complete, it would not be possible to install ToT WebKit onto devices running OS builds which do not contain this switcheroo code change. So, the shorter the migration, the more painful it will be.

The right thing to do is add metadata to roots that tells `darwinup` to remove the XPC services upon installation. This avoids all the problem above, because we can remove the pre-installed XPC services as we install the new root. We have precedent for this, and since root building is an internal workflow, the changes needed will be outside of the WebKit repo.

**Comment by pvollan** [comment]
This change is no longer needed since we are skipping install of XPC services in this configuration.

**Pull request closed without merging**